### PR TITLE
core/dotCMS#20110 Add timezone component to Push and Publish dialog

### PR DESCRIPTION
**Pull request by alfredo-dotcms**
### Proposed Changes
* We show the “current time zone” (based on the user's browsers)
* If user clicks it
* Show the dropdown to change it (filled with a list of timezones used in BE)

### Checklist
- [ ] Tests
- [ ] Translations
- [ ] Security Implications Contemplated (add notes if applicable)

### Additional Info
** issue https://github.com/dotCMS/core/issues/20110 **
BE branch `issue-20110-push-publishing-is-not-respecting-time-zones-and-is-basing-publish-time-on-the-browser-time`

### Screenshots
Original             |  Updated
:-------------------------:|:-------------------------:
** original screenshot **  |  ** https://www.figma.com/file/xyjX828sx4rbAdjOB06dTG/DotCMS?node-id=664%3A10773 **

### CODE REVIEW

File: apps/dotcms-ui/src/app/view/components/_common/forms/dot-push-publish-form/dot-push-publish-form.component.ts [hunk 1]
```diff
@@ -109,6 +130,20 @@ export class DotPushPublishFormComponent
         );
     }
 
+    private loadTimezones(): void {
+        this.dotTimezoneService
+            .get()
+            .pipe(take(1))
+            .subscribe((timezones: DotTimeZone[]) => {
+                this.timeZoneOptions = timezones.map((item: DotTimeZone) => {
```
Comment (fmontes) [inline apps/dotcms-ui/src/app/view/components/_common/forms/dot-push-publish-form/dot-push-publish-form.component.ts:138]: Can this be an Observable and use async pipe?

**Review by fmontes — CHANGES_REQUESTED** [review]

File: apps/dotcms-ui/src/app/view/components/_common/forms/dot-push-publish-form/dot-push-publish-form.component.html [hunk 2]
```diff
@@ -72,6 +72,20 @@
             ></dot-field-validation-message>
         </div>
     </div>
+    <div class="p-field push-publish-dialog__timezone-label">
+        <span>{{ localTimezone }}</span> -
+        <a href="#" (click)="toggleTimezonePicker($event)"> {{ changeTimezoneActionLabel }}</a>
```
Comment (hmoreras) [inline apps/dotcms-ui/src/app/view/components/_common/forms/dot-push-publish-form/dot-push-publish-form.component.html:77]: use the `dm` pipe

Comment (hmoreras) [inline apps/dotcms-ui/src/app/view/components/_common/forms/dot-push-publish-form/dot-push-publish-form.component.html:77]: use the `dm` pipe

File: apps/dotcms-ui/src/app/view/components/_common/forms/dot-push-publish-form/dot-push-publish-form.component.spec.ts [hunk 3]
```diff
@@ -162,6 +171,53 @@ describe('DotPushPublishFormComponent', () => {
...
+        expect(timeZoneLabel.outerText).toEqual(
+            pushPublishForm.timeZoneOptions.find(({ value }) => value === localTZ)['label']
+        );
+    });
+
+    it('should display TimeZones dropdown when "Change" link clicked', () => {
+        const changeTZLink = fixture.debugElement.query(
+            By.css('.push-publish-dialog__timezone-label a')
+        ).nativeElement;
+        changeTZLink.click();
```
Comment (hmoreras) [inline apps/dotcms-ui/src/app/view/components/_common/forms/dot-push-publish-form/dot-push-publish-form.component.spec.ts:195]: `triggerEventHandler` here ?

File: apps/dotcms-ui/src/app/view/components/_common/forms/dot-push-publish-form/dot-push-publish-form.component.ts [hunk 4]
```diff
@@ -84,6 +91,20 @@ export class DotPushPublishFormComponent
         this.value.emit(this.form.value);
     }
 
+    updateTimezoneLabel(timezone: string): void {
+        this.localTimezone = this.timeZoneOptions.find(({ value }) => value === timezone)['label'];
+        debugger
```
Comment (hmoreras) [inline apps/dotcms-ui/src/app/view/components/_common/forms/dot-push-publish-form/dot-push-publish-form.component.ts:96]: remove

File: apps/dotcms-ui/src/app/view/components/_common/forms/dot-push-publish-form/dot-push-publish-form.component.ts [hunk 5]
```diff
@@ -84,6 +91,20 @@ export class DotPushPublishFormComponent
         this.value.emit(this.form.value);
     }
 
+    updateTimezoneLabel(timezone: string): void {
```
Comment (hmoreras) [inline apps/dotcms-ui/src/app/view/components/_common/forms/dot-push-publish-form/dot-push-publish-form.component.ts:94]: doc

File: apps/dotcms-ui/src/app/view/components/_common/forms/dot-push-publish-form/dot-push-publish-form.component.ts [hunk 6]
```diff
@@ -84,6 +91,20 @@ export class DotPushPublishFormComponent
...
+        this.localTimezone = this.timeZoneOptions.find(({ value }) => value === timezone)['label'];
+        debugger
+    }
+
+    toggleTimezonePicker(event: MouseEvent): void {
```
Comment (hmoreras) [inline apps/dotcms-ui/src/app/view/components/_common/forms/dot-push-publish-form/dot-push-publish-form.component.ts:99]: doc

**Review by hmoreras — CHANGES_REQUESTED** [review]

**Review by ivorpad — DISMISSED** [review]

File: apps/dotcms-ui/src/app/view/components/_common/forms/dot-push-publish-form/dot-push-publish-form.component.ts (hunk 1 above)
Comment (alfredo-dotcms) [reply apps/dotcms-ui/src/app/view/components/_common/forms/dot-push-publish-form/dot-push-publish-form.component.ts:138]: I don't think so, the data needs to be transformed and then that transformed data will be used as a reference to functions `updateTimezoneLabel` and `setUsersTimeZone`

File: apps/dotcms-ui/src/app/view/components/_common/forms/dot-push-publish-form/dot-push-publish-form.component.html (hunk 2 above)
Comment (alfredo-dotcms) [reply apps/dotcms-ui/src/app/view/components/_common/forms/dot-push-publish-form/dot-push-publish-form.component.html:77]: it is not possible `changeTimezoneActionLabel`  changes when link clicked
https://github.com/dotCMS/core-web/blob/2793c9b3a59d65a66273fd149925c78c924e3a99/apps/dotcms-ui/src/app/view/components/_common/forms/dot-push-publish-form/dot-push-publish-form.component.ts#L101

File: apps/dotcms-ui/src/app/view/components/_common/forms/dot-push-publish-form/dot-push-publish-form.component.spec.ts (hunk 3 above)
Comment (alfredo-dotcms) [reply apps/dotcms-ui/src/app/view/components/_common/forms/dot-push-publish-form/dot-push-publish-form.component.spec.ts:195]: nah

**Review by fmontes — APPROVED** [review]

**Review by hmoreras — APPROVED** [review]

**Review by fmontes — APPROVED** [review]

**Pull request merged**